src/wifi.rs

```diff
@@ -403,6 +411,15 @@ impl EspWifi {
             esp!(unsafe { esp_netif_attach_wifi_station(netif.1) })?;
             esp!(unsafe { esp_wifi_set_default_wifi_sta_handlers() })?;
 
+            if let Some(hn) = hostname {
+                if let Ok(hostname) = CString::new(hn) {
+                    esp!(unsafe { esp_netif_set_hostname(netif.1, hostname.as_ptr()) })?;
```

Comment:
Minor nit, but I suggest you surface two new public members in the ESP-IDF Netif wrapper, `EspNetif`:
* `get_hostname(&self)`
* `set_hostname(&self...)`

... and then call `set_hostname` here.

You can model `get_hostname` by roughly copy-pasting the code from the existing `get_name` function. For `set_name` I think you've figured it out already how it should look (roughly).

Reply:
Yeah that looks like a better fit. But then shouldn't I call `set_hostname` at `EspNetif::new`?